### ISSUE TITLE
Child slide updates don't update slider

### ISSUE THREAD

**Issue by scouttyg** [akiran/react-slick#100]
It was mentioned here: https://github.com/akiran/react-slick/issues/29#issuecomment-126169829, but I wanted to re-emphasize the need for this functionality.

Here's the use case:

Imagine you have a "flashcard" application, which might look something like this (you'll have to imagine what the rest might look like, but this should give you an idea of what I'm talking about):

```html
<select>
  <option></option>
  <option>History</option>
  <option>Science</option>
  <option>English</option>
</select>

<Slider>
  {questionsNode}
</Slider>
```

The idea is, on page load, you want all flashcards to show up from all topics, or you could choose an option from the select dropdown, and the slider then only shows those flashcards.

This would be solved with ```componentWillRecieveProps```, and this code did exist before (see: https://github.com/akiran/react-slick/commit/01e6e4b60adb61a1282bf358f77241f2e0efa861) but was removed due to this issue: https://github.com/akiran/react-slick/issues/85

For me, I feel like removing componentWillRecieveProps is a worse way of doing this -- ideally, in issue 85, there would be some code written to manage the state of the current slide (perhaps using the initialSlide option to update the settings?)

**Comment by akiran** [comment]
Yes, I agree, componentWillRecieveProps should be there.

Need to fix #85 in a different way

**Comment by gkilian** [comment]
I was just gonna file a ticket for this until I saw this. This is a must fix!

**Comment by akiran** [comment]
@scouttyg, @gkilian  Released a fix in 0.6.8. checkout
